runtimebp/cpu.go

```diff
@@ -127,11 +131,83 @@ func defaultMaxProcsFormula(n float64) int {
 	return int(math.Ceil(n))
 }
 
-// GOMAXPROCS sets runtime.GOMAXPROCS with the default formula,
+func scaledMaxProcsFormula(n float64) int {
+	i := fetchMaxProcsMult()
+	return int(math.Ceil(n * float64(i)))
+}
+
+func fetchMaxProcsMult() int {
+	// Allow using a multiplier for number of processes relative to limit.
+	// Not catching the ok here because this function will never be called unless
+	// the environment variable is set
+	v, _ := os.LookupEnv("GOMAXPROCSMULT")
+	i, err := strconv.Atoi(v)
+	if err != nil {
+		fmt.Fprintf(
+			os.Stderr,
+			"runtimebp.fetchMaxProcsMult: GOMAXPROCSMULT is set to %q, which is invalid, falling back to 2 to ensure parallelism: %v",
+			v,
+			err,
+		)
+		i = 2
```

Comment:
So, the reason I defaulted the multiplier to 2 is that if we have it set to 1 we can literally never burst above request, and that seems like it doesn't really align with the desired state for default behavior.  It'd function identically to having a limit in that case.  Does that make sense?

Reply:
there are two different "bursting" here.

one is os/container level cpu bursting, that is controlled by cpu limits, and has nothing to do with GOMAXPROCS

the other is the actual "parallelism" of the go program, which is controlled by GOMAXPROCS, but since we have an overall minimal of 2 (that's after all the multiplications we did), I don't think having a default multiplier set to 2 is meaningful.

Reply:
So, the thing is, if GOMAXPROCS is set to 2, you can use at most 2 CPU cores.  We actually had an incident yesterday where this came up.  It's functionally the same as having the CPU limit actually set to 2.

Reply:
ok then :)